### DISCUSSION TITLE
Recursive response model field is not populated if it is set to child instance, but annotated with parent model class

### DISCUSSION THREAD

**Discussion by arturb90** [fastapi/fastapi#8814] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import fastapi

from fastapi import FastAPI
from pydantic import BaseModel
from typing import List
from typing import Dict


class Baz(BaseModel):
    pass


class Foo(Baz):
    count: int


class Spam(BaseModel):
    baz: Baz


fastapi_app = FastAPI()


@fastapi_app.get(
    "/api/spam",
    response_model=Spam,
)
async def spam():
    m = Spam(baz=Foo(count=1))
    print(m.dict())
    return m
```

### Description

* Call endpoint `/api/spam`.
* Expected output: `{'baz': {'count': 1}}`
* Actual output: `{'baz': {}}`

Pydantic output is fine:
```python3
m = Spam(baz=Foo(count=1))
print(m.dict())
# {'baz': {'count': 1}}
```
This works also:
```python3
class Spam(BaseModel):
    baz: Foo
```

### Operating System

macOS

### FastAPI Version

0.68.1

### Python Version

Python 3.9.2

**Comment by NinaHwang** [comment]
That's because of the response model. From response model's point of view, Baz always contains nothing.

**Comment by arturb90** [comment]
Is this expected behaviour or a bug? I would have assumed that the missing attribute can be resolved since pydantic is correctly resolving it.

**Comment by YuriiMotov** [comment, marked as answer]
Yes, this is expected behaviour.

You can use `Union` to make it working as you want:
```python
class Spam(BaseModel):
    baz: Union[Foo, Baz]
```

Notice, that more specific types (in this case `Foo`) should go first in `Union`
